internal/vendors/accessor.go

```diff
@@ -35,101 +35,99 @@ func (p *postgresVendorAccessor) GetSomeStuff(ctx context.Context) ([]string, er
 	return results, nil
 }
 
-func (p *postgresVendorAccessor) GetAll(ctx context.Context, spec database.PaginationSpec) (*AccessorGetAllPaginationData, error) {
-	args := database.BuildPaginationArgs(spec)
-
-	dataQuery := `SELECT 
-		"id",
-		"name",
-		"description",
-		"bp_id",
-		"bp_name",
-		"rating",
-		"area_group_id",
-		"area_group_name",
-		"sap_code",
-		"modified_date",
-		"modified_by",
-		"dt" 
-		FROM vendor
-		ORDER BY created_at $1
-		LIMIT $2
-		OFFSET $3
-		`
-
-	rows, err := p.db.Query(dataQuery, args.Order, args.Limit, args.Offset)
-	if err != nil {
-		return nil, err
+func (p *postgresVendorAccessor) GetAll(ctx context.Context, spec GetAllVendorSpec) (*AccessorGetAllPaginationData, error) {
+	paginationArgs := database.BuildPaginationArgs(spec.PaginationSpec)
+
+	// Initialize clauses and arguments
+	var (
+		joinClauses     []string
+		whereClauses    []string
+		extraClauses    []string
+		args            []interface{}
+		i               = 1
+		extraClausesRaw = []string{
+			"LIMIT $%d",
+			"OFFSET $%d",
+		}
+	)
+
+	// Build WHERE clause for location
+	if spec.Location != "" {
+		whereClauses = append(whereClauses, fmt.Sprintf("area_group_name = $%d", i))
+		args = append(args, spec.Location)
+		i++
 	}
 
-	defer rows.Close()
+	// Build JOIN and WHERE clauses for product
+	if spec.Product != "" {
+		joinClauses = append(joinClauses, "JOIN product_vendor pv ON pv.vendor_id = v.id")
+		joinClauses = append(joinClauses, "JOIN product p ON p.id = pv.product_id")
 
-	vendors := []Vendor{}
-
-	for rows.Next() {
-		var vendor Vendor
-		err := rows.Scan(
-			&vendor.ID,
-			&vendor.Name,
-			&vendor.Description,
-			&vendor.BpID,
-			&vendor.BpName,
-			&vendor.Rating,
-			&vendor.AreaGroupID,
-			&vendor.AreaGroupName,
-			&vendor.SapCode,
-			&vendor.ModifiedDate,
-			&vendor.ModifiedBy,
-			&vendor.Date,
-		)
-		if err != nil {
-			return nil, err
+		productNameList := strings.Fields(spec.Product)
+		for _, word := range productNameList {
+			whereClauses = append(whereClauses, fmt.Sprintf("p.name iLIKE $%d", i))
+			args = append(args, "%"+word+"%")
+			i++
 		}
-		vendors = append(vendors, vendor)
 	}
 
-	if err := rows.Err(); err != nil {
-		return nil, err
+	// Set order by default value
+	if spec.OrderBy == "" {
+		spec.OrderBy = "v.dt"
+	} else {
+		spec.OrderBy = "v." + spec.OrderBy
 	}
```

Comment:
This handler could be moved to ```BuildPaginationArgs``` on helper if we agree ```orderBy``` to be inside pagiantionSpec

Reply:
It would be risky to generalize it, since not all tables has dt and in this case we also use "v." prefix. I think it's better to handle it per use case, so it would customizable.